### PR TITLE
Extracted API to its own repo

### LINKED ISSUE #95: Make API a library on its own
Allows for other apps to use the API.

### PR DESCRIPTION
**Pull request by boginw**
This closes #95. See [this](https://github.com/aau-giraf/api_client) to keep up with the API client. This PR contains only deletions and updated references, so it should be pretty quick.

I'd like to comment, that the only thing that was done to make the tests pass which mock the API (which was the reason why this PR was reverted), was to update the imports of those files (since `package:weekplanner/providers/api/*` no longer exists)

### CODE REVIEW

**Review by ralled96 — CHANGES_REQUESTED** [review]
The 'new' API returns an exception instead of null when there is no weekplans. 
As we talked about on Slack, this is fixed by handling the exception:

On line 46, in weekplan_bloc.dart it is fixed by adding this:
`_api.week.getNames(_user.id).listen(_weekNameModelsList.add, 
      onError: (Object exception) => getAllWeekInfo(null));`

**Comment by boginw** [comment]
@ralled96 this has now been fixed

**Review by ralled96 — DISMISSED** [review]

File: lib/screens/weekplan_screen.dart [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
 import 'package:weekplanner/di.dart';
-import 'package:weekplanner/models/activity_model.dart';
-import 'package:weekplanner/models/enums/weekday_enum.dart';
-import 'package:weekplanner/models/week_model.dart';
 import 'package:weekplanner/widgets/giraf_app_bar_widget.dart';
 import 'package:weekplanner/widgets/pictogram_image.dart';
 
 /// <summary>
-/// The WeekplandScreen is used to display a week 
+/// The WeekplandScreen is used to display a week
```
Comment (TobiasPalludan) [inline lib/screens/weekplan_screen.dart]:
```suggestion
/// The WeekplanScreen is used to display a week
```

File: lib/models/response_model.dart [hunk 2]
```diff
@@ -1,36 +0,0 @@
-import 'package:weekplanner/models/enums/error_key.dart';
```
Comment (TobiasPalludan) [inline lib/models/response_model.dart:1]: This file was deleted, but it wasn't added to the api_client repository. Is that on purpose?

Comment (boginw) [reply lib/models/response_model.dart:1]: https://github.com/aau-giraf/api_client/blob/master/lib/models/enums/error_key.dart

**Review by TobiasPalludan — APPROVED** [review]

**Review by ralled96 — APPROVED** [review]

**Pull request merged**